### PR TITLE
[e2e tests] Add support for external sites in CI

### LINKED ISSUE #48920: Add support for external test sites in ci-jobs
We need to run tests against external sites, but the ci-jobs and ci.yml only support wp-env at the moment. We need to be able to configure, parse the config and send it to ci.yml (via env variables?)

### PR DESCRIPTION
**Pull request by adimoldovan**
### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/48920
Part of pdV5qK-wB-p2

Add support for running e2e tests against any site in CI.

- added scripts that encrypt and decrypt .env files.
- added a new env entry - `default-pressable` to validate the setup.
- configured the `Core e2e tests - default Pressable site` job to run on `daily-checks`.

### How to test the changes in this Pull Request:

Locally:

Run the tests with `pnpm test:e2e:with-env default-pressable`. You'll need to find and set the `E2E_ENV_KEY` for this.
Expected:
- The `tests/e2e-pw/envs/default-pressable/.env.enc` is decrypted into `tests/e2e-pw/.env`
- The `tests/e2e-pw/envs/default-pressable/env-setup.sh` script is executed. It doesn't do anything (yet)
- The tests configured in `tests/e2e-pw/envs/default-pressable/playwright.config.js` should run

In CI:
Check the `Core e2e tests - default Pressable site` job: https://github.com/woocommerce/woocommerce/actions/runs/9782549328/job/27009409289?pr=49017. It was temporarily configured to run for PRs to validate the CI setup.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
           build: 'false'
 
       - uses: './.github/actions/setup-woocommerce-monorepo'
-        if: ${{ github.ref_type != 'tag' }}
+        if: ${{ github.ref_type != 'tag' &&  matrix.testEnv.shouldCreate }}
```
Comment (kalessil) [inline .github/workflows/ci.yml:124]: Nice! Do we have the means to skip the step for PHP unit tests (to try)?

File: plugins/woocommerce/tests/e2e-pw/bin/dotenv.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  echo "Usage: $0 [-e|-d] env-name"
+  exit 1
+fi
+
+SCRIPT_PATH=$(cd "$(dirname "${BASH_SOURCE[0]}")" || return; pwd -P)
+encFile="$SCRIPT_PATH/../envs/$1/.env.enc"
+decFile="$SCRIPT_PATH/../.env"
+
+if [ "$operation" == "encrypt" ]; then
+  openssl enc -aes-256-cbc -iter 1000 -pass env:E2E_ENV_KEY -in "$decFile" -out "$encFile"
```
Comment (kalessil) [inline plugins/woocommerce/tests/e2e-pw/bin/dotenv.sh:35]: `E2E_ENV_KEY` needed for local testing.

**Review by kalessil — APPROVED** [review]
LGTM! If possible @adimoldovan, can we extract readme update in a separate PR where I can spend some reviewing it and not blocking the main changes?

**Comment by adimoldovan** [comment]
> LGTM! If possible @adimoldovan, can we extract readme update in a separate PR where I can spend some reviewing it and not blocking the main changes?

Thanks for reviewing @kalessil. You can comment on the readme here and we can address it in a follow-up PR.

File: plugins/woocommerce/tests/e2e-pw/bin/dotenv.sh (hunk 2 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/bin/dotenv.sh:35]: You can find it the Secret Store.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adimoldovan) [reply .github/workflows/ci.yml:124]: I'm not sure there is a case where we want it skipped for unit tests.

**Comment by kalessil** [comment]
> You can comment on the readme here and we can address it in a follow-up PR.

Works for me, thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kalessil) [reply .github/workflows/ci.yml:124]: I can evaluate it in a separate PR, but looking into build commands, WooCommerce build steps are around building assets. I assume PHPUnit will not need them to run tests.

Comment (kalessil) [reply .github/workflows/ci.yml:124]: I'll draft a PR today to test my proposal - let's skip it for this PR to move faster.

Comment (kalessil) [reply .github/workflows/ci.yml:124]: Drafted https://github.com/woocommerce/woocommerce/pull/49135, marking this thread as resolved.

**Pull request merged**